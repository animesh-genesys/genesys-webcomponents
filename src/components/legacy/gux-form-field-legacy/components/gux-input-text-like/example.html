<form onchange="notify(event)" oninput="notify(event)">
  <h1>gux-form-field-legacy (Text like)</h1>

  <p>
    slot a input[type=email] / input[type=password] / input[type=search] /
    input[type=text] element into a gux-form-field-legacy element
  </p>

  <div>
    <h2>Label Position Above</h2>
    <gux-form-field-legacy label-position="above">
      <input
        slot="input"
        name="email"
        type="email"
        placeholder="joe.bloggs@example.com"
      />
      <label slot="label">Email</label>
    </gux-form-field-legacy>

    <gux-form-field-legacy label-position="above">
      <input
        slot="input"
        name="text"
        type="text"
        placeholder="Enter a string"
      />
      <label slot="label">Text</label>
      <span slot="error">Required</span>
    </gux-form-field-legacy>
  </div>

  <h2>Clearable</h2>

  <div>
    <gux-form-field-legacy clearable label-position="above">
      <input
        slot="input"
        name="email-clearable"
        type="email"
        placeholder="joe.bloggs@example.com"
        value="daragh@example.com"
      />
      <label slot="label">Email</label>
    </gux-form-field-legacy>

    <gux-form-field-legacy clearable label-position="above">
      <input
        slot="input"
        name="password-clearable"
        type="password"
        placeholder="Enter a password"
      />
      <label slot="label">Password</label>
    </gux-form-field-legacy>

    <gux-form-field-legacy clearable label-position="above">
      <input
        slot="input"
        name="text-clearable"
        type="text"
        placeholder="Enter a string"
      />
      <label slot="label">Text</label>
    </gux-form-field-legacy>
  </div>

  <h2>Disabled</h2>

  <div>
    <gux-form-field-legacy clearable label-position="above">
      <input
        slot="input"
        name="email-clearable"
        type="email"
        placeholder="joe.bloggs@example.com"
        value="me@example.com"
        disabled
      />
      <label slot="label">Email</label>
    </gux-form-field-legacy>

    <gux-form-field-legacy clearable label-position="above">
      <input
        slot="input"
        name="password-clearable"
        type="password"
        placeholder="Enter a password"
        value="password"
        disabled
      />
      <label slot="label">Password</label>
    </gux-form-field-legacy>

    <gux-form-field-legacy clearable label-position="above">
      <input
        slot="input"
        name="text-clearable"
        type="text"
        placeholder="Enter a string"
        value="text"
        disabled
      />
      <label slot="label">Text</label>
      <span slot="error" id="test1">Required</span>
    </gux-form-field-legacy>
  </div>

  <div>
    <h2>Label Position Beside</h2>
    <gux-form-field-legacy label-position="beside">
      <input
        slot="input"
        name="email"
        type="email"
        placeholder="joe.bloggs@example.com"
      />
      <label slot="label">Email</label>
    </gux-form-field-legacy>

    <gux-form-field-legacy label-position="beside">
      <input
        slot="input"
        name="text"
        type="text"
        placeholder="Enter a string"
      />
      <label slot="label">Text</label>
      <span slot="error">Required</span>
    </gux-form-field-legacy>
  </div>

  <h2>Clearable</h2>

  <div>
    <gux-form-field-legacy clearable label-position="beside">
      <input
        slot="input"
        name="email-clearable"
        type="email"
        placeholder="joe.bloggs@example.com"
        value="daragh@example.com"
      />
      <label slot="label">Email</label>
    </gux-form-field-legacy>

    <gux-form-field-legacy clearable label-position="beside">
      <input
        slot="input"
        name="password-clearable"
        type="password"
        placeholder="Enter a password"
      />
      <label slot="label">Password</label>
    </gux-form-field-legacy>

    <gux-form-field-legacy clearable label-position="beside">
      <input
        slot="input"
        name="text-clearable"
        type="text"
        placeholder="Enter a string"
      />
      <label slot="label">Text</label>
    </gux-form-field-legacy>
  </div>

  <h2>Disabled</h2>

  <div>
    <gux-form-field-legacy clearable label-position="beside">
      <input
        slot="input"
        name="email-clearable"
        type="email"
        placeholder="joe.bloggs@example.com"
        value="me@example.com"
        disabled
      />
      <label slot="label">Email</label>
    </gux-form-field-legacy>

    <gux-form-field-legacy clearable label-position="beside">
      <input
        slot="input"
        name="password-clearable"
        type="password"
        placeholder="Enter a password"
        value="password"
        disabled
      />
      <label slot="label">Password</label>
    </gux-form-field-legacy>

    <gux-form-field-legacy clearable label-position="beside">
      <input
        slot="input"
        name="text-clearable"
        type="text"
        placeholder="Enter a string"
        value="text"
        disabled
      />
      <label slot="label">Text</label>
      <span slot="error">Required</span>
    </gux-form-field-legacy>
  </div>

  <h2>Visually Hidden Label</h2>
  <gux-form-field-legacy clearable label-position="screenreader">
    <input
      slot="input"
      name="text-clearable"
      type="text"
      placeholder="Enter a string"
    />
    <label slot="label">This will be read by a screenreader</label>
  </gux-form-field-legacy>

  <h2>With a smaller width</h2>

  <gux-form-field-legacy clearable label-position="above" style="width: 120px">
    <input
      slot="input"
      name="text-clearable"
      type="text"
      placeholder="Enter a string"
      value="Text"
    />
    <label slot="label">Text</label>
    <span slot="error">Required</span>
  </gux-form-field-legacy>
</form>
