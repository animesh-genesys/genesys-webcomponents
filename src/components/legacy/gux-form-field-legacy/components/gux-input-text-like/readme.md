# gux-input-text-like

<!-- Auto Generated Below -->


## Properties

| Property    | Attribute   | Description | Type      | Default     |
| ----------- | ----------- | ----------- | --------- | ----------- |
| `clearable` | `clearable` |             | `boolean` | `undefined` |


## Slots

| Slot      | Description                                                             |
| --------- | ----------------------------------------------------------------------- |
| `"input"` | Required slot for input[type="text" \| type="email" \| type="password"] |


## Dependencies

### Used by

 - [gux-dropdown-legacy](../../../gux-dropdown-legacy)
 - [gux-form-field-legacy](../..)

### Depends on

- [gux-icon](../../../../stable/gux-icon)

### Graph
```mermaid
graph TD;
  gux-input-text-like --> gux-icon
  gux-dropdown-legacy --> gux-input-text-like
  gux-form-field-legacy --> gux-input-text-like
  style gux-input-text-like fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
