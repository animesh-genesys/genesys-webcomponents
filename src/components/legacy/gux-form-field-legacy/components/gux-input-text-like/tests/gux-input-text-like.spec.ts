import { newSpecPage, SpecPage } from '@stencil/core/testing';
import MutationObserver from 'mutation-observer';

import { GuxInputTextLike } from '../gux-input-text-like';

const components = [GuxInputTextLike];
const language = 'en';

describe('gux-input-text-like', () => {
  let page: SpecPage;

  beforeEach(async () => {
    global.MutationObserver = MutationObserver;

    page = await newSpecPage({
      components,
      html: `
        <gux-input-text-like>
          <input type="text" slot="input"/>
        </gux-input-text-like>
      `,
      language
    });
  });

  it('should build', async () => {
    expect(page.rootInstance).toBeInstanceOf(GuxInputTextLike);
  });

  it('should render', async () => {
    expect(page.root).toMatchSnapshot();
  });
});
