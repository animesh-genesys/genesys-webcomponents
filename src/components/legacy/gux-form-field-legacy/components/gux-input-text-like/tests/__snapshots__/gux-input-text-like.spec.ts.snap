// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-input-text-like should render 1`] = `
<gux-input-text-like>
  <!---->
  <div class="gux-input-container">
    <input slot="input" type="text">
  </div>
</gux-input-text-like>
`;
